test/events.test.js

```diff
@@ -2,59 +2,177 @@
 import test from 'tape';
 import mapboxgl from 'mapbox-gl-js-mock';
 import GLDraw from '../';
-import { accessToken, createMap, features } from './utils';
+import { accessToken, createMap, features, click } from './utils';
+import spy from 'sinon/lib/sinon/spy'; // avoid babel-register-related error by importing only spy
 
 mapboxgl.accessToken = accessToken;
 
+const EVENT_TIME = 32;
+
 var feature = features.point;
 
 var map = createMap();
+spy(map, 'fire');
 
-var timeout = function(t, time, msg) {
-  var end = t.end.bind(t);
+var Draw = GLDraw();
+map.addControl(Draw);
 
-  var done = false;
+test('draw.modified is fired when placing a point', t => {
+  Draw.deleteAll();
+  Draw.changeMode('draw_point');
+  map.fire.reset();
+  click(map, {
+    originalEvent: {
+        isShiftKey: false,
+        stopPropagation: function() {}
+      },
+      point: {x: 0, y:0},
+      lngLat: {lng: 0, lat: 0}
+  });
 
-  setTimeout(function() {
-    if(t.notDone()) {
-      done = true;
-      t.fail(msg || 'Timed out');
-      t.end();
-    }
-  })
+  t.equal('mousedown', map.fire.getCall(0).args[0]);
+  t.equal('mouseup', map.fire.getCall(1).args[0]);
+  t.equal('draw.modechange', map.fire.getCall(2).args[0]);
+  t.equal('draw.modified', map.fire.getCall(3).args[0]);
+  t.end();
+});
+
+test('draw.modified is fired when drawing a line_string', t => {
+  Draw.deleteAll();
+  Draw.changeMode('draw_line_string');
+  map.fire.reset();
+  click(map, {
+    originalEvent: {
+        isShiftKey: false,
+        stopPropagation: function() {}
+      },
+      point: {x: 0, y:0},
+      lngLat: {lng: 0, lat: 0}
+  });
 
-  t.notDone = function() {
-    return !done;
-  }
+  click(map, {
+    originalEvent: {
+        isShiftKey: false,
+        stopPropagation: function() {}
+      },
+      point: {x: 5, y:0},
+      lngLat: {lng: 5, lat: 0}
+  });
 
-  t.end = function() {
-    done = true;
-    end();
-  }
+  click(map, {
+    originalEvent: {
+        isShiftKey: false,
+        stopPropagation: function() {}
+      },
+      point: {x: 0, y:5},
+      lngLat: {lng: 0, lat: 5}
+  });
 
-  return t;
-}
+  var events = [
+    'mousedown',
+    'mouseup',
```

Comment:
`draw.modified` deliberately does not fire after the first vertex is created, right?